clients/destination.go

```diff
@@ -21,6 +40,59 @@ func NewDestinationClient(cc grpc.ClientConnInterface) *DestinationClient {
 	}
 }
 
+// NewSourceClientSpawn starts a new source plugin process, connects to it via gRPC server
+// and returns a new SourceClient
+func NewDestinationClientSpawn(ctx context.Context, path string, opts ...DestinationClientOption) (*DestinationClient, error) {
```

Comment:
```suggestion
func NewDestinationClient(ctx context.Context, path string, opts ...DestinationClientOption) (*DestinationClient, error) {
```

Naming is hard, but I'm not sure the current name is a great description of what the function does... Maybe `NewDestinationClientFromPath`? With further clarification in the comment, I think it should be fairly clear.

Reply:
Or maybe `SpawnNewDestinationClient` 🤔

Reply:
Im good with either, I thought about all three and this one seemed to me the most consistent but Im good with either

Reply:
What about `NewManagedDestinationClient`, because it's managing the underlying process for you?